### ISSUE TITLE
push notifications

### ISSUE THREAD

**Issue by GallagherCommaJack** [kalix-systems/herald#27]

**Comment by mobile-bungalow** [comment]
This is a complex one, QML supports push notifications for MacOs, but uses a homespun-looking version for Linux desktop environments.  Baeo mentioned wanting notificationd support. which means a long with PSSSt. we may be writing a cross platform push notification wrapper for libherald. For IOs this also includes codesigning.

**Comment by GallagherCommaJack** [comment]
Yeah this is going to be more important for iOS/Android than desktop because the expectations are different. I'm probably fine with QT's default solution for Linux.

For Android we'll probably just use Google firebase. We could also look into using mqtt, since Android will let us keep the connection open and this is kinda what it's built for.

On iOS we'll have to just do whatever Apple demands - though we can maybe use a service like pushy if we don't want to deal with their API for some reason.

**Comment by BaeoMaltinsky** [comment]
Update: We have push notifications for new messages on Linux and macOS, and support for standard desktop BSD can be added by tweaking a conditional compilation attribute. I believe that the library we're using, [`notify-rust`](https://crates.io/crates/notify-rust) has Windows support on a branch and I can update the code to support when its merged.
